src/main/java/frc/robot/subsystems/SwerveDrivetrain.java

```diff
@@ -127,7 +116,8 @@ public SwerveDrivetrain(SwerveIdConfig frontLeftConfig, SwerveIdConfig frontRigh
         this.frontLeft.getSwerveMotor().getDriveMotor().setInverted(Constants.SWERVE_MODULE_PROFILE.isFrontLeftInverted());
         this.backRight.getSwerveMotor().getDriveMotor().setInverted(Constants.SWERVE_MODULE_PROFILE.isBackRightInverted());
         this.backLeft.getSwerveMotor().getDriveMotor().setInverted(Constants.SWERVE_MODULE_PROFILE.isBackLeftInverted());
-        this.poseEstimator = new SwervePosEstimator(encodedSwerveSparkMaxFL,encodedSwerveSparkMaxFR,encodedSwerveSparkMaxBL,encodedSwerveSparkMaxBR,kinematics,getGyro());
+        gyro.updateIfDisabled();
```

Comment:
Maybe not the best idea - we can add "synchronousGet" to make sure we have a reading is necessary.

Reply:
now I just get the first reading in the constructor of the Gyro. This is not an important value anywasy because we reset pose anyways.